base/abstractarraymath.jl

```diff
@@ -386,6 +386,34 @@ end
 
 end#module
 
+struct EachSlice{E,M,A,I,L} <: AbstractArray{E,M}
```

Comment:
I love that this is an `AbstractArray`. :)

I note that the majority of our concrete array types tend to be `XxxArray`, so as a suggestion we could name this as `SlicedArray`?

Reply:
What's the advantage of being distinct from `SubArray` with `Base.Slice` indices?

Reply:
This means something different?  SubArrays iterate over their elements; this iterates over whole slices (SubArrays themselves).  I don't think you can bend a SubArray by itself into this behavior.

Reply:
Sorry for the noise, I had a misunderstanding.